apps/passport-client/src/dispatch.ts

```diff
@@ -1128,12 +1106,14 @@ async function handleAgreedPrivacyNotice(
   update: ZuUpdate,
   version: number
 ): Promise<void> {
-  await saveSelf({ ...state.self, terms_agreed: version });
-  update({
-    self: { ...state.self, terms_agreed: version },
-    loadedIssuedPCDs: false,
-    modal: { modalType: "none" }
-  });
+  if (state.self) {
+    saveSelf({ ...state.self, terms_agreed: version });
```

Comment:
definitely very wary of the changes that remove `await`

Reply:
`saveSelf` isn't async either.